hack/build.sh

```diff
@@ -323,7 +324,7 @@ display_help() {
     cat <<EOT
 Knative client build script
 
-Usage: $(basename $BASH_SOURCE) [... options ...]
+Usage: $(basename "${BASH_SOURCE[@]}") [... options ...]
```

Comment:
I don't think that makes much sense, as this script is never meant to be sourced  by another script (which only then makes a difference to the original version)

Reply:
So I would keep the original version or use "${BASH_SOURCE[0]}" if this makes shellcheck happier